bin/cheritest/cheritest_vm.c

```diff
@@ -265,6 +267,66 @@ cheritest_vm_shm_open_anon_unix_surprise(const struct cheri_test *ctp __unused)
 	}
 }
 
+#ifdef CHERIABI_TESTS
+
+/*
+ * We can fork processes with shared file descriptor tables, including
+ * shared access to a kqueue, which can hoard capabilities for us, allowing
+ * them to flow between address spaces.  It is difficult to know what to do
+ * about this case, but it seems important to acknowledge.
+ */
+void
+cheritest_vm_cap_share_fd_kqueue(const struct cheri_test *ctp __unused)
+{
+	int kq, pid;
+
+	kq = CHERITEST_CHECK_SYSCALL(kqueue());
+	pid = fork();
```

Comment:
kqueue's are not shared across `fork()`.  This is implemented in `fdcopy()` which doesn't copy over file descriptors if they can't be passed across a unix domain socket (`DFLAG_PASSABLE`), and kqueue's can't be passed over a socket.  I would expect the `kevent()` syscall in the child to fail with `EBADF`.  However, you could use `rfork(RFPROC)` in the test to work around that.  For CheriABI we could just refuse to implement some of the rfork() variants I think.  Using rfork() for Linux-threads isn't a common thing anymore, and there doesn't seem to be any other really good reasons to use RFPROC without one of the fd flags.

Reply:
Indeed; I'd originally written it with `rfork` and then changed and hadn't tested again.  To prevent future me from making the same mistake, I've added a printout.

Restricting `rfork` seems like a viable option here.  Is it safe for CheriABI to refuse `RFSIGSHARE` too?

Reply:
I think restricting rfork() might be ok.  It's not a portable interface.  I wonder if we can actually outright remove some of the variants in stock FreeBSD.

Reply:
As much as I love the flexibility of `rfork`, that is probably worthwhile.  In the interim, any further suggestions for these commits? :)